### PR TITLE
Add 'archived' field for reuse model and endpoints

### PR DESCRIPTION
**Pull request by bolinocroustibat**
Closes https://github.com/datagouv/data.gouv.fr/issues/1273

Also:
- Minor .gitignore update
- Minor translation typo fix

### CODE REVIEW

File: udata/core/reuse/api.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ class ReuseListAPI(API):
     @api.marshal_with(reuse_page_fields)
     def get(self):
         args = reuse_parser.parse()
-        reuses = Reuse.objects(deleted=None, private__ne=True)
+        reuses = Reuse.objects(archived=None, deleted=None, private__ne=True)
```
Comment (maudetes) [inline udata/core/reuse/api.py:104]: For now, we can keep returning archived reuses on list and suggest endpoints

File: udata/core/reuse/models.py [hunk 2]
```diff
@@ -113,7 +114,7 @@ def is_visible(self):
 
     @property
     def is_hidden(self):
-        return len(self.datasets) == 0 or self.private or self.deleted
+        return len(self.datasets) == 0 or self.private or self.deleted or self.archived
```
Comment (maudetes) [inline udata/core/reuse/models.py:117]: Same here, I would not modify it before having discussed product strategy towards archived reuses visibility.

**Review by maudetes — COMMENTED** [review]
Thank you for this PR! For I think we shouldn't filter out archived reuses yet :)

Don't forget the changelog entry ;)

File: udata/core/reuse/api.py (hunk 1 above)
Comment (bolinocroustibat) [reply udata/core/reuse/api.py:104]: Thanks, that makes sense. Fixed

File: udata/core/reuse/models.py (hunk 2 above)
Comment (bolinocroustibat) [reply udata/core/reuse/models.py:117]: Indeed, fixed!

File: udata/core/reuse/search.py [hunk 3]
```diff
@@ -43,7 +44,7 @@ def is_indexable(cls, reuse):
 
     @classmethod
     def mongo_search(cls, args):
-        reuses = Reuse.objects(deleted=None, private__ne=True)
+        reuses = Reuse.objects(archived=None, deleted=None, private__ne=True)
```
Comment (maudetes) [inline udata/core/reuse/search.py:47]: Woops, here too
```suggestion
        reuses = Reuse.objects(deleted=None, private__ne=True)
```

File: udata/core/reuse/api.py [hunk 4]
```diff
@@ -254,7 +254,7 @@ class ReusesSuggestAPI(API):
     def get(self):
         '''Reuses suggest endpoint using mongoDB contains'''
         args = suggest_parser.parse_args()
-        reuses = Reuse.objects(deleted=None, private__ne=True, title__icontains=args['q'])
+        reuses = Reuse.objects(archived=None, deleted=None, private__ne=True, title__icontains=args['q'])
```
Comment (maudetes) [inline udata/core/reuse/api.py:257]:
```suggestion
        reuses = Reuse.objects(deleted=None, private__ne=True, title__icontains=args['q'])
```

**Review by maudetes — APPROVED** [review]
Looks good to me!

I think we can prioritize https://github.com/datagouv/data.gouv.fr/issues/1393 next to add this functionality in the admin.

File: js/views/reuse.vue [hunk 5]
```diff
@@ -194,6 +205,24 @@ export default {
...
+      archived_reuse.archived = new Date().toISOString();
+      API.reuses.update_reuse({reuse: this.reuse.id, payload: archived_reuse},
+          (response) => {
+              this.reuse.on_fetched(response);
+          }
+      );
+    },
+    unarchive() {
+      const archived_reuse = this.reuse;
+      archived_reuse.archived = null;
```
Comment (ThibaudDauce) [inline js/views/reuse.vue:219]: Why not modify the `this.reuse` directly?

File: js/views/reuse.vue [hunk 6]
```diff
@@ -194,6 +205,24 @@ export default {
     edit() {
       this.$go({ name: 'reuse-edit', params: { oid: this.reuse.id } })
     },
+    archive() {
+      const archived_reuse = this.reuse;
+      archived_reuse.archived = new Date().toISOString();
```
Comment (ThibaudDauce) [inline js/views/reuse.vue:210]: Why not modify the `this.reuse` directly?

**Review by ThibaudDauce — CHANGES_REQUESTED** [review]
:+1: 

Could you add a test for getting archived reuses? I know I will modify this code in my PR https://github.com/opendatateam/udata/pull/3066 so it could be cool to have a test checking this new feature :-)

**Comment by bolinocroustibat** [comment]
> 👍
> 
> Could you add a test for getting archived reuses? I know I will modify this code in my PR #3066 so it could be cool to have a test checking this new feature :-)

OK I'll try! Not sure I know how frontend tests work yet though :)

**Comment by bolinocroustibat** [comment]
> > 👍
> > Could you add a test for getting archived reuses? I know I will modify this code in my PR #3066 so it could be cool to have a test checking this new feature :-)
> 
> OK I'll try! Not sure I know how frontend tests work yet though :)

Since we want the API to return even archived reuses, we cannot test on the API return, but added a test for archived model field behaviour. LMKWYT

**Review by ThibaudDauce — APPROVED** [review]

File: js/views/reuse.vue [hunk 7]
```diff
@@ -194,6 +205,22 @@ export default {
     edit() {
       this.$go({ name: 'reuse-edit', params: { oid: this.reuse.id } })
     },
+    archive() {
+      this.reuse.archived = new Date().toISOString();
```
Comment (maudetes) [inline js/views/reuse.vue:209]: Reacting late on this PR, but I think we would like to add a watch on `reuse.archived` to show a label when archived (similarly to datasets) for a feedback to the user on the action.

**Pull request merged**